XenModel/Actions/Network/NetworkingActionHelpers.cs

```diff
@@ -269,6 +269,45 @@ private static void ReconfigureManagement_(AsyncAction action, PIF pif, int hi)
             action.Description = string.Format(Messages.ACTION_CHANGE_NETWORKING_MANAGEMENT_RECONFIGURED, pif.Name);
         }
 
+        internal static void PoolReconfigureManagement(AsyncAction action, PIF up_pif, PIF down_pif, int hi)
+       {
+           System.Diagnostics.Trace.Assert(down_pif.host.opaque_ref == up_pif.host.opaque_ref);
+
+           int lo = action.PercentComplete;
+           int inc = (hi - lo) / 3;
+           lo += inc;
+           PoolManagementReconfigure_( action, up_pif, lo);
+
+           /* Pool recover slaves spwans two tasks dbsync (update_env) and server_init on each slaves. 
+            * Only after their completion master will be able to execute reconfigure_IPs. 
+            * Hence, we allow some setteling time for them to complete.
+            */
+           Thread.Sleep(10000);

```

Comment:
I checked the tasks for pool_management_reconfigure, it completes after pool_recover_slaves gets completed. Only after pool_recover_slaves is completed these two tasks are spawned so its difficult to track them since these are not sub tasks of pool_recover_slaves. Even I was not keen in giving a timeout value, however in case if we have to check for some pool property we would require to poll (busy loop) for some fixed amount of time until the property is changed, so ultimately the timeout factor comes into play.